### PR TITLE
Add feature to read log from opensearch

### LINKED ISSUE #33619: Add a compatibility with Opensearch search for log management
### Description

Dear Airflow team,

Today, it is possible to configure Airflow to read logs of the tasks from ElasticSearch.

Is it possible to add the support of Opensearch as source of the task logs ?

Thank you.

### Use case/motivation

Log sinks are often used in large companies. Concentrate logs in a log sinks is very practical for searches.

Opensearch and Elasticsearch have similar methods, so it should not create major change.

ElasticSearch has added some checks in the client library, so it is not possible anymore to Opensearch with the elasticsearch library.

Cloud providers such as OVHCloud or Scaleway, propose Opensearch as log manager as a service.

Opensearch is a full opensource solution contrary to Elasticsearch.

### Related issues

No

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Owen-CH-Leung**
Closes #33619 

Airflow currently only supports reading remote log from elasticsearch. This PR adds feature to also allow reading remote logs from opensearch.

Similar to reading remote logs from elasticsearch, users need to use other tools to import logs into Opensearch. 

The set up is very similar to setting up remote logging for Elasticsearch. In the airflow.cfg: 

```
remote_logging = True
remote_log_conn_id = opensearch_default

[opensearch]
host = [OS host name] (e.g. my-opensearch-01)
port = [port number] (e.g. 9200)
username = 
password = 
```

Depending on your Opensearch config, you may also add the following in airflow.cfg :

```
[opensearch_configs]
http_compress = False
use_ssl = False
verify_certs = False
ssl_assert_hostname = False
ssl_show_warn = False
ca_certs = 
```

If you set up everything successfully (And correctly ship your logs to Opensearch), webserver should output logs like the following: 

```
[2024-08-27T15:02:54.611+0000] {base.py:258} INFO - POST http://my-opensearch-01:9200/_all/_count [status:200 request:0.034s]
[2024-08-27T15:02:54.654+0000] {base.py:258} INFO - POST http://my-opensearch-01:9200/_all/_search?size=1000&sort=log&from=0 [status:200 request:0.040s]
192.168.65.1 - - [27/Aug/2024:15:02:54 +0000] "GET /api/v1/dags/consume_1_or_2_with_dataset_expressions/dagRuns/manual__2024-08-27T15:00:00.024040+00:00/taskInstances/consume_1_or_2_with_dataset_expressions/logs/1?full_content=false HTTP/1.1" 200 4219 "http://localhost:28080/dags/consume_1_or_2_with_dataset_expressions/grid?dag_run_id=manual__2024-08-27T15%3A00%3A00.024040%2B00%3A00&task_id=consume_1_or_2_with_dataset_expressions&tab=logs"
```

### CODE REVIEW

**Comment by Owen-CH-Leung** [comment]
@eladkal can I seek your review for this PR ? Thanks

**Review by eladkal — APPROVED** [review]

**Comment by eladkal** [comment]
@kaxil @ephraimbuddy can you review the core parts of this PR? Given the work for Airflow 3 I am not sure about the provider<->core log integrations. To my understanding it is still coupled with core to some extant

File: airflow/providers/opensearch/provider.yaml [hunk 1]
```diff
@@ -59,3 +59,144 @@ operators:
...
+          The port number of Opensearch host
+        version_added: 1.5.0
+        type: integer
+        example: ~
+        default: ""
+      username:
+        description: |
+          The username for connecting to Opensearch
+        version_added: 1.5.0
+        type: string
```
Comment (ephraimbuddy) [inline airflow/providers/opensearch/provider.yaml:88]:
```suggestion
        type: string
        sensitive: true
```

File: airflow/providers/opensearch/provider.yaml [hunk 2]
```diff
@@ -59,3 +59,144 @@ operators:
...
+        example: ~
+        default: ""
+      password:
+        description: |
+          The password for connecting to Opensearch
+        version_added: 1.5.0
+        type: string
```
Comment (ephraimbuddy) [inline airflow/providers/opensearch/provider.yaml:96]:
```suggestion
        type: string
        sensitive: true
```

**Comment by eladkal** [comment]
Noting: I set Airflow 3 milestone for this PR.
The provider part will be released in next wave after it's merged but for the core parts we can't release it in 2.10.x as this is feature, we can't also release it in 2.11 as this is a feature that does not serve the porpuse of bridge release as we agreed (reference to https://github.com/apache/airflow/blob/main/dev/README_AIRFLOW3_DEV.md#developing-for-airflow-3-and-210x--211x )

File: airflow/providers/opensearch/provider.yaml (hunk 1 above)
Comment (Owen-CH-Leung) [reply airflow/providers/opensearch/provider.yaml:88]: Thanks. Added `sensitive: true`

File: airflow/providers/opensearch/provider.yaml (hunk 2 above)
Comment (Owen-CH-Leung) [reply airflow/providers/opensearch/provider.yaml:96]: Thanks. Added `sensitive: true`

File: docs/apache-airflow-providers-opensearch/logging/index.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+.. _write-logs-opensearch:
+
+Writing logs to Opensearch
+-----------------------------
+
```
Comment (eladkal) [inline docs/apache-airflow-providers-opensearch/logging/index.rst:22]:
```suggestion

*Added in provider version 1.5.0*
Availably only with Airflow>=3.0
```

**Pull request merged**